File: 가장먼노드.py
from collections import deque

def solution(n, edge):
    graph = [([0]*n) for _ in range(n)]
    visitied = [False]*n

    for i in edge:
        graph[i[0]-1][i[1]-1] = 1
        graph[i[1]-1][i[0]-1] = 1

    def BFS(start):
        queue = deque()
        visitied[start] = True 
        answer = [0] * n  

        queue.append((start, 0))

        while queue:
            node, dist = queue.popleft()
            for indx, isConnected in enumerate(graph[node]):
                if visitied[indx] == False and isConnected == 1:
                    visitied[indx] = True
                    queue.append((indx, dist + 1))
                    answer[indx] = dist + 1 

        max_dist = max(answer)
        return answer.count(max_dist)

    return BFS(0)

File: 가장먼노드.cpp
#include <iostream>;
#include <vector>;
#include <queue>;


using namespace std;

int solution(int n, vector<vector<int>> edge){

    vector<vector<int>> graph(n+1);
    vector<bool> visited(n+1, false);
    
    for(const auto& next : edge ){
        graph[next[0]].push_back(next[1]);
        graph[next[1]].push_back(next[0]); 
    }

}

int bfs(vector<vector<int>>&  graph, vector<bool> visited, int start){
    queue<pair<int,int>> q;
    visited[1] = true;

    int answer = 0;
    int lef = 0;
    int max = 0;

    for(const auto& next : graph[start]){
        q.push({next,1});
        visited[next] = true;
    }

    while(!q.empty()){
        auto [node, dist] = q.front();
        q.pop();

        for(const auto& a : graph[node]){
            if(visited[a] != true){
                visited[a] = true;
                q.push({a,dist+1});
            }
        }

        if (dist > max) {
            max = dist;
            answer = 1;
        } else if (dist == max) {
            answer++;
        }
    }

    return answer;


}